demo/app/services/optimizely_service.rb

```diff
@@ -0,0 +1,80 @@
+require 'logger'
+require 'optimizely'
+require 'sinatra'
+
+class OptimizelyService
+  @@optimizely_client = nil
+  @@variation = nil
+  def initialize(datafile)
+    @datafile = datafile
+    @errors = []
+    @optimizely_client = @@optimizely_client
+    @variation = @@variation
+  end
+
+  def self.optimizely_client_present?
+    @@optimizely_client.present?
+  end
+  
+  def instantiate!
+    @logger = SinatraLogger.new(Logger.new(STDOUT))
+    @@optimizely_client = Optimizely::Project.new(
+        @datafile,
+        Optimizely::EventDispatcher.new,
```

Comment:
I think you can pass null instead of  Optimizely::EventDispatcher.new and NoOpErrorHandler, and Optimizely will auto create these. 
Check if Optimizely does this. 
Params should be (@datafile, null, @logger). Please verify

Reply:
Done